sdk/src/accounts/customizedCadenceBulkAccountLoader.ts

```diff
@@ -0,0 +1,150 @@
+import { BulkAccountLoader } from './bulkAccountLoader';
+import { Commitment, Connection, PublicKey } from '@solana/web3.js';
+
+export class CustomizedCadenceBulkAccountLoader extends BulkAccountLoader {
+	private customIntervalIds: Map<string, NodeJS.Timeout>;
+	private customPollingGroups: Map<number, Set<string>>;
+	private defaultPollingFrequency: number;
+
+	constructor(
+		connection: Connection,
+		commitment: Commitment,
+		defaultPollingFrequency: number
+	) {
+		super(connection, commitment, defaultPollingFrequency);
+		this.customIntervalIds = new Map();
+		this.customPollingGroups = new Map();
+		this.defaultPollingFrequency = defaultPollingFrequency;
+	}
+
+	private updateCustomPolling(frequency: number): void {
+		const frequencyStr = frequency.toString();
+		const existingInterval = this.customIntervalIds.get(frequencyStr);
+		if (existingInterval) {
+			clearInterval(existingInterval);
+			this.customIntervalIds.delete(frequencyStr);
+		}
+
+		const group = this.customPollingGroups.get(frequency);
+		if (group && group.size > 0) {
+			const intervalId = setInterval(async () => {
+				const accounts = Array.from(group)
+					.map((key) => this.accountsToLoad.get(key))
+					.filter((account) => account !== undefined);
+
+				if (accounts.length > 0) {
+					await this.loadChunk([accounts]);
+				}
+			}, frequency);
+			this.customIntervalIds.set(frequencyStr, intervalId);
+		}
+	}
+
+	public setCustomPollingFrequency(
+		publicKey: PublicKey,
+		newFrequency: number
+	): void {
+		const key = publicKey.toBase58();
+
+		// Remove from old frequency group
+		for (const [frequency, group] of this.customPollingGroups.entries()) {
+			if (group.has(key)) {
+				group.delete(key);
+				if (group.size === 0) {
+					const intervalId = this.customIntervalIds.get(frequency.toString());
+					if (intervalId) {
+						clearInterval(intervalId);
+						this.customIntervalIds.delete(frequency.toString());
+					}
+					this.customPollingGroups.delete(frequency);
+				}
+				this.updateCustomPolling(frequency);
+				break;
+			}
+		}
+
+		// Add to new frequency group
+		let group = this.customPollingGroups.get(newFrequency);
+		if (!group) {
+			group = new Set();
+			this.customPollingGroups.set(newFrequency, group);
+		}
+		group.add(key);
+
+		this.updateCustomPolling(newFrequency);
+	}
+
+	public async addAccount(
+		publicKey: PublicKey,
+		callback: (buffer: Buffer, slot: number) => void,
+		customPollingFrequency?: number
+	): Promise<string> {
+		const id = await super.addAccount(publicKey, callback);
```

Comment:
when we add the account to the parent BulkAccountLoader class, and this is happening for all accounts regardless of the customPollingFrequency group they are meant to be in, does that mean they are all being fetched during the default polling too?